### ISSUE TITLE
[Feature]: App developer can start an application

### ISSUE THREAD

**Issue by Birdrock** [cloudfoundry/cf-k8s-controllers#172]
### Blockers/Dependencies

* Eirini LRPs will be created in another issue: https://github.com/cloudfoundry/cf-k8s-controllers/issues/72

### Background

**As an** App Developer
**I want** to be able to start my application
**So that** my application workloads are run

## Scenarios
---
### Happy Path

**GIVEN** my app exists with a droplet assigned, and a desiredState of `STOPPED`
**AND** it has associated process(es) that are scaled to zero
**WHEN I** issue a start command
```
curl "https://api.example.org/v3/apps/<guid>/actions/start" \
  -X POST \
  -H "Authorization: bearer [token]"
```

**THEN I** get a response similar to the following
```
HTTP/1.1 200 OK
Content-Type: application/json

{
  "guid": "1cb006ee-fb05-47e1-b541-c34179ddc446",
  "name": "my_app",
  "state": "STARTED",
  "created_at": "2016-03-17T21:41:30Z",
  "updated_at": "2016-03-18T11:32:30Z",
  "lifecycle": {
    "type": "buildpack",
    "data": {
      "buildpacks": ["java_buildpack"],
      "stack": "cflinuxfs3"
    }
  },
  "relationships": {
    "space": {
      "data": {
        "guid": "2f35885d-0c9d-4423-83ad-fd05066f8576"
      }
    }
  },
  "links": {
    "self": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446"
    },
    "space": {
      "href": "https://api.example.org/v3/spaces/2f35885d-0c9d-4423-83ad-fd05066f8576"
    },
    "processes": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/processes"
    },
    "packages": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/packages"
    },
    "environment_variables": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/environment_variables"
    },
    "current_droplet": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/droplets/current"
    },
    "droplets": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/droplets"
    },
    "tasks": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/tasks"
    },
    "start": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/actions/start",
      "method": "POST"
    },
    "stop": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/actions/stop",
      "method": "POST"
    },
    "revisions": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/revisions"
    },
    "deployed_revisions": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/revisions/deployed"
    },
    "features": {
      "href": "https://api.example.org/v3/apps/1cb006ee-fb05-47e1-b541-c34179ddc446/features"
    }
  },
  "metadata": {
    "labels": {},
    "annotations": {}
  }
}
```
**AND** the desired state of the CFApp is updated
**AND** workloads pods are created (for now just change desiredState)

---
### Unhappy Path Sad Path: Droplet isn't assigned

**GIVEN**  I have the shim running and cf-k8s-controllers installed and deployed to my K8s cluster
**AND** an App **does exist** but **no** Build/Droplet assigned 
**WHEN I** make the following request

```bash
curl "https://api.example.org/v3/apps/<guid>/actions/start" \
  -X POST \
  -H "Authorization: bearer [token]"
```

**THEN I:** get back the following response

```json
HTTP/1.1 422 Unprocessable Entity

{
  "errors": [
    {
      "detail": "Assign a droplet before starting this app.'",
      "title": "CF-UnprocessableEntity",
      "code": 10008
    }
  ]
}
```

### Dev Notes
- For now just change the desiredState from STOPPED -> STARTED as long as it has a droplet assigned

**Comment by tcdowney** [comment]
Closing: This was addressed in https://github.com/cloudfoundry/cf-k8s-controllers/issues/315.
